docs/custom-plugins/visitor.md

```diff
@@ -0,0 +1,111 @@
+# Visitor pattern
+
+_Visitor_ is a design pattern that allows operations to be performed on individual elements in a more complex structure. Redocly uses it as the basis for custom plugins where every node in a document must be evaluated, for example when applying rules or decorators.
+
+To understand how this applies to your API description, think of the document as a tree structure. The top level elements are entries like `info` and `components`. To examine the `description` field in the `info` section, the visitor goes to the `info` node first, then on to the `description` node. This pattern is repeated all over the document as the visitor pattern works its way around the document tree.
+
+<a id="visitor-object"></a><a id="format-of-visitor-object"></a>
+
+## Structure of the visitor object
+
+In your plugin, create a JavaScript visitor object, and describe the functionality required for each type of node.
+
+Redocly CLI calls `enter()` while going down the tree and `leave()` going up the tree after processing the node and its children.
+If the `skip()` predicate is defined and returns `true` the node is ignored for this visitor.
```

Comment:
question: Does this result in all children nodes being ignored too? I'm trying to understand the reason to skip something versus omitting it from the plugin.